File: day07/part1.swift
var total = 0
while let line = readLine() {
    let pieces = line.split(separator: ": ")
    let target = Int(pieces[0])!
    let numbers = pieces[1].split(separator: " ").map { Int($0)! }

    if (matches(target: target, numbers: numbers, i: 1, currentValue: numbers[0])) {
        total += target
    }
}


func matches(target: Int, numbers: [Int], i: Int, currentValue: Int) -> Bool {
    if (i == numbers.count) {
        return currentValue == target
    }

    return matches(target: target, numbers: numbers, i: i + 1, currentValue: currentValue + numbers[i])
        || matches(target: target, numbers: numbers, i: i + 1, currentValue: currentValue * numbers[i])
}

print(total)

File: day07/part1.js
let total = 0;
for await (const line of console) {
    const pieces = line.split(": ");
    const target = Number(pieces[0]);
    const numbers = pieces[1].split(" ").map((n) => Number(n));

    if (matches(target, numbers, 1, numbers[0])) {
        total += target;
    }
}

console.log(total);

function matches(target, numbers, i, currentValue) {
    if (i === numbers.length) {
        return currentValue === target;
    }

    return matches(target, numbers, i + 1, currentValue + numbers[i])
        || matches(target, numbers, i + 1, currentValue * numbers[i]);
}